SU2_CFD/src/variables/CNEMONSVariable.cpp

```diff
@@ -102,6 +104,14 @@ bool CNEMONSVariable::SetPrimVar(unsigned long iPoint, CFluidModel *FluidModel)
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
     DiffusionCoeff(iPoint, iSpecies) = Ds[iSpecies];
 
+  su2double T   =  Primitive(iPoint,nSpecies);
+  su2double Tve =  Primitive(iPoint,nSpecies+1);
+
+  su2double* val_eves = GetEve(iPoint);
+  hs = fluidmodel->ComputeSpeciesEnthalpy(T, Tve, val_eves);
```

Comment:
2)
```suggestion
  const auto& hs = fluidmodel->ComputeSpeciesEnthalpy(T, Tve, val_eves);
```
Like this there is no copy.
And similarly for the others I mentioned.